deepchecks/tabular/context.py

```diff
@@ -30,16 +33,118 @@
+    y_pred_test: np.ndarray
+        Array of the model prediction over the test dataset.
+    y_proba_train: np.ndarray
+        Array of the model prediction probabilities over the train dataset.
+    y_proba_test: np.ndarray
+        Array of the model prediction probabilities over the test dataset.
+    """
+
+    train: pd.DataFrame
+    test: pd.DataFrame
+    ind_map: t.Dict[str, t.Dict[t.Any, int]]
+    y_pred: t.Dict[str, np.ndarray]
+    y_proba: t.Dict[str, np.ndarray]
+
+    def __init__(self,
+                 train: Dataset,
+                 test: Dataset,
+                 y_pred_train: np.ndarray,
+                 y_pred_test: np.ndarray,
+                 y_proba_train: np.ndarray,
+                 y_proba_test: np.ndarray,):
+        self.ind_map = defaultdict(dict)
+
+        if train is not None and test is not None:
+            # check if datasets have same indexes
+            if set(train.data.index) & set(test.data.index):
+                train.data.index = map(lambda x: f'train-{x}', list(train.data.index))
+                test.data.index = map(lambda x: f'test-{x}', list(test.data.index))
+                warnings.warn('train and test datasets have common index - adding "train"/"test"'
+                              ' prefixes. To avoid that provide datasets with no common indexes '
+                              'or don\'t use the static predictions feature.')
+
+        if train is not None:
+            self.train = train.features_columns
+            for i, ind in enumerate(self.train.index):
+                self.ind_map[DatasetKind.TRAIN][ind] = i
+
+        if test is not None:
+            self.test = test.features_columns
+            for i, ind in enumerate(self.test.index):
+                self.ind_map[DatasetKind.TEST][ind] = i
+
+        if y_pred_train is not None or y_pred_test is not None:
+            self.y_pred = {}
+            self.y_pred[DatasetKind.TRAIN] = y_pred_train
+            self.y_pred[DatasetKind.TEST] = y_pred_test
+            self.predict = self._predict
+
+        if y_proba_train is not None or y_proba_train is not None:
+            self.y_proba = {}
+            self.y_proba[DatasetKind.TRAIN] = y_proba_train
+            self.y_proba[DatasetKind.TEST] = y_proba_test
+            self.predict_proba = self._predict_proba
+
+    def _get_dataset(self, dataset_kind: DatasetKind):
+        if dataset_kind == DatasetKind.TRAIN:
+            return self.train
+        if dataset_kind == DatasetKind.TEST:
+            return self.test
+        raise DeepchecksValueError(f'Unexpected dataset kind {dataset_kind}')
+
+    def _validate_data(self, data: pd.DataFrame, dataset_kind: DatasetKind):
+        sub_index = list(data.index)[:10]
+        return (data.loc[sub_index].fillna('nan') ==
+                self._get_dataset(dataset_kind).loc[sub_index].fillna('nan')).all().all()
+
+    def _get_dataset_kind(self, data: pd.DataFrame):
```

Comment:
I have a slight concern that this constant computation of set intersection can become costly when the data is big, and we expect to do this a lot (every time we do predict on any dataset). 
Can we have something that will enable us not to do all this in case we get the whole dataset to predict on? I expect that a bit % of operations will be "just give me the predictions for the whole train/test".

Reply:
That is why it is a tiny sample

Reply:
What is a tiny sample? Can you elaborate?

Reply:
I tend to agree with @matanper, matching 10 samples may not be fool-proof enough.